apps/vx-scan/frontend/src/api.ts

```diff
@@ -33,6 +52,67 @@ export const getMachineConfig = {
   },
 } as const;
 
+export const getAuthStatus = {
+  queryKey(): QueryKey {
+    return ['getAuthStatus'];
+  },
+  useQuery() {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(), () => apiClient.getAuthStatus(), {
+      refetchInterval: AUTH_STATUS_POLLING_INTERVAL_MS,
+    });
+  },
+} as const;
+
+export const checkPin = {
+  useMutation() {
+    const apiClient = useApiClient();
+    const queryClient = useQueryClient();
+    return useMutation(apiClient.checkPin, {
+      async onSuccess() {
+        // Because we poll auth status with high frequency, this invalidation isn't strictly
+        // necessary
+        await queryClient.invalidateQueries(getAuthStatus.queryKey());
+      },
+    });
+  },
+} as const;
+
+export const readCardData = {
+  queryKey(): QueryKey {
+    return ['readCardData'];
+  },
+  useQuery<T>(schema: CardDataSchemaName) {
+    const apiClient = useApiClient();
+    return useQuery(
+      this.queryKey(),
+      () =>
+        // In the conversion from the Api type to the ApiClient type, generics are lost, so we cast
+        // back to the original type here
+        (apiClient.readCardData as Api['readCardData'])<T>({ schema }),
```

Comment:
So in addition to the Zod schema serialization challenge affecting type safety, there's a separate issue of the conversion from the `Api` type to the `ApiClient` type causing generics to be lost.

You can see that here when I remove the cast:

<img width="960" alt="error" src="https://user-images.githubusercontent.com/12616928/217305629-36d726dd-96e1-45cc-8db8-c17a9f00fbe9.png">

Haven't gotten to the bottom of it yet, but something about this utility is having that effect:

https://github.com/votingworks/vxsuite/blob/46b0c7ec6da8b39f46eec25b8133e48272cec097/libs/grout/src/client.ts#L8-L13

The following:

```ts
grout.AsyncRpcMethod<(<T>({ schema }: {
    schema: "ScannerReportDataSchema";
}) => Promise<Result<Optional<T>, Error | SyntaxError | ZodError<any>>>)
```

Becomes:

```ts
(args_0: {
    schema: "ScannerReportDataSchema";
}) => Promise<Result<unknown, Error | SyntaxError | ZodError<any>>>
```

Reply:
👍 Irrelevant for now given what I landed on